XiEditor/Theme.swift

```diff
@@ -0,0 +1,125 @@
+    let selection: NSColor
+    /// text color of the selection regions.
+    let selectionForeground: NSColor
+    /// Color of the selection regions border.
+    let selectionBorder: NSColor?
+    /// Background color of inactive selections (inactive view).
+    let inactiveSelection: NSColor?
+    /// Text color of inactive selections (inactive view).
+    let inactiveSelectionForeground: NSColor?
+
+    /// The color of the shadow used when a text area can be horizontally scrolled.
+    let shadow: NSColor?
+}
+
+extension Theme {
+    static func defaultTheme() -> Theme {
+        return Theme(foreground: NSColor.black,
+              background: NSColor.white,
+              caret: NSColor.black,
+              lineHighlight: nil,
+              findHighlight: NSColor(deviceWhite: 0.8, alpha: 0.4),
+              findHighlightForeground: nil,
+              gutter: NSColor(deviceWhite: 0.9, alpha: 1.0),
+              gutterForeground: NSColor(deviceWhite: 0.5, alpha: 1.0),
+              selection: NSColor.selectedTextBackgroundColor,
+              selectionForeground: NSColor.selectedTextColor,
+              selectionBorder: nil,
+              inactiveSelection: NSColor(colorLiteralRed: 0.8, green: 0.8, blue: 0.8, alpha: 1.0),
+              inactiveSelectionForeground: NSColor.selectedTextColor,
+              shadow: nil
+            )
+    }
+
+    init(jsonObject dict: [String: AnyObject]) {
+        let foreground = NSColor(jsonRgbaColor: dict["foreground"] as? [String: AnyObject] ?? [:])
+        let background = NSColor(jsonRgbaColor: dict["background"] as? [String: AnyObject] ?? [:])
+        let caret = NSColor(jsonRgbaColor: dict["caret"] as? [String: AnyObject] ?? [:])
+        let line_highlight = NSColor(jsonRgbaColor: dict["line_highlight"] as? [String: AnyObject] ?? [:])
+
+        let find_highlight = NSColor(jsonRgbaColor: dict["find_highlight"] as? [String: AnyObject] ?? [:])
+        let find_highlight_foreground = NSColor(jsonRgbaColor: dict["find_highlight_foreground"] as? [String: AnyObject] ?? [:])
+        let gutter = NSColor(jsonRgbaColor: dict["gutter"] as? [String: AnyObject] ?? [:])
+        let gutter_foreground = NSColor(jsonRgbaColor: dict["gutter_foreground"] as? [String: AnyObject] ?? [:])
+
+        let selection = NSColor(jsonRgbaColor: dict["selection"] as? [String: AnyObject] ?? [:])
+        let selection_foreground = NSColor(jsonRgbaColor: dict["selection_foreground"] as? [String: AnyObject] ?? [:])
+        let selection_border = NSColor(jsonRgbaColor: dict["selection_border"] as? [String: AnyObject] ?? [:])
+        let inactive_selection = NSColor(jsonRgbaColor: dict["inactive_selection"] as? [String: AnyObject] ?? [:])
+        let inactive_selection_foreground = NSColor(jsonRgbaColor: dict["inactive_selection_foreground"] as? [String: AnyObject] ?? [:])
+        let shadow = NSColor(jsonRgbaColor: dict["shadow"] as? [String: AnyObject] ?? [:])
+
+        let defaults = Theme.defaultTheme()
```

Comment:
Yea, there are a few things here that are a bit fudgy, because the client doesn't know the themes on first launch. The default values are currently (mostly) just copied from the `InspiredGithub` theme, for no good reason. I'm not sure how defaults should actually work, e.g. for the gutter view, which isn't specified in a lot of theme files but can be pretty jarring if it's a mismatch.

I'm very happy to have people being nitty about this stuff, basically. 😎